lib/src/native/realm_core.dart

```diff
@@ -181,6 +182,85 @@ class _RealmCore {
+  }
+
+  static void request_callback(Object userData, realm_http_request request, Pointer<Void> request_context) {
+    () async { // TODO: Use another isolate, perhaps?
+      final client = userData as HttpClient;
+      client.connectionTimeout = Duration(milliseconds: request.timeout_ms);
+      try {
+        final url = Uri.parse(request.url.cast<Utf8>().toDartString());
+        late HttpClientRequest mappedRequest;
+        switch (request.method) {
+          case 4: // DELETE
+            mappedRequest = await client.deleteUrl(url);
+            break;
+          case 3: // PUT
+            mappedRequest = await client.putUrl(url);
+            break;
+          case 2: // PATCH
+            mappedRequest = await client.patchUrl(url);
+            break;
+          case 1: // POST
+            mappedRequest = await client.postUrl(url);
+            break;
+          case 0: // GET
+          default:
+            mappedRequest = await client.getUrl(url);
+            break;
+        }
+
+        final body = request.body.asTypedList(request.body_size); // we avoid a potentially expensive copy here
+        mappedRequest.add(body);
+
+        // Unfortunately we need to copy the headers, due to the interface of HttpClientRequest
+        for (int i = 0; i < request.num_headers; ++i) {
+          final header = request.headers[i];
+          final name = header.name.cast<Utf8>().toDartString();
+          final value = header.value.cast<Utf8>().toDartString();
+          mappedRequest.headers.add(name, value);
+        }
+
+        // Do the call..
+        final response = await mappedRequest.close();
+        final responseBody = await response.fold<List<int>>([], (acc, l) => acc..addAll(l)); // gather response
+
+        // Report back to core
+        using((arena) {
+          final response_pointer = arena<realm_http_response>();
+          final responseRef = response_pointer.ref;
+
+          responseRef.body = responseBody.toInt8Ptr(arena); // need to copy here :-(
+          responseRef.body_size = responseBody.length;
+
+          int headerCnt = 0;
+          response.headers.forEach((name, values) {
+            // Freaking odd interface :-/
+            headerCnt += values.length;
+          });
+
+          responseRef.headers = arena<realm_http_header>(headerCnt);
+          responseRef.num_headers = headerCnt;
+
+          response.headers.forEach((name, values) {
+            int idx = 0;
+            for (final value in values) {
+              final headerRef = responseRef.headers.elementAt(idx).ref;
+              headerRef.name = name.toUtf8Ptr(arena);
+              headerRef.value = value.toUtf8Ptr(arena);
+            }
+          });
+
+          _realmLib.realm_http_transport_complete_request(request_context, response_pointer);
+        });
+      } catch (ex) {
+        print(ex); // TODO!
```

Comment:
We shall avoid printing here. Perhaps we need to report an error to core with completing the request with the correct response_pointer. I suspect we need to complete the request on an error.

Reply:
I'll discuss with core team, if we can call complete with response_pointer set to null in this case.

Reply:
Discussed with @RedBeard0531. We can leave all but custom_error nulled out.